### PR TITLE
Construct transformer layers from attention and feedforward layers

### PR DESCRIPTION
**Pull request by danieldk**
## Description

Let the `EncoderLayer` and `DecoderLayer` constructor take `SelfAttention` and `PointwiseFeedForward` as arguments, rather than a large number of configuration options. This provides more structure to constructing a transformer layer.

Similarly, remove `RotaryEmbeddingConfig` and let the `SelfAttention` constructor take `QueryKeyRotaryEmbeddings`.

### Types of change

Maintenance.

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: curated_transformers/models/bert/encoder.py [hunk 1]
```diff
@@ -5,7 +5,15 @@
 from torch import Tensor
 from torch.nn import LayerNorm
 
-from ...layers.attention import AttentionMask, QkvHeadSharing, QkvMode
+from curated_transformers.layers.feedforward import PointwiseFeedForward
```
Comment (shadeMe) [inline curated_transformers/models/bert/encoder.py:8]: Relative import.

File: curated_transformers/models/falcon/layer.py [hunk 2]
```diff
@@ -4,12 +4,13 @@
 from torch import Tensor
 from torch.nn import Module
 
+from curated_transformers.layers.embeddings import QueryKeyRotaryEmbeddings
```
Comment (shadeMe) [inline curated_transformers/models/falcon/layer.py:7]: Relative import.

File: curated_transformers/models/llama/decoder.py [hunk 3]
```diff
@@ -5,13 +5,11 @@
 from torch import Tensor
 from torch.nn import Dropout, Embedding, ModuleList
 
-from ...layers.attention import (
-    AttentionMask,
-    QkvHeadSharing,
-    QkvMode,
-    RotaryEmbeddingConfig,
-)
+from curated_transformers.layers.embeddings import QueryKeyRotaryEmbeddings
```
Comment (shadeMe) [inline curated_transformers/models/llama/decoder.py:8]: Relative import.

File: curated_transformers/models/roberta/encoder.py [hunk 4]
```diff
@@ -5,7 +5,9 @@
 from torch import Tensor
 from torch.nn import LayerNorm
 
-from ...layers.attention import AttentionMask, QkvHeadSharing, QkvMode
+from curated_transformers.layers.feedforward import PointwiseFeedForward
```
Comment (shadeMe) [inline curated_transformers/models/roberta/encoder.py:8]: Relative import.

File: curated_transformers/layers/transformer.py [hunk 5]
```diff
@@ -59,85 +52,34 @@ class _TransformerLayer(Module):
...
-            Activation used by the feed-forward layers.
-            Applied on the intermediate representation.
-            See :py:class:`curated_transformers.layers.feedforward.PointwiseFeedForward`
-            for supported activations.
+        :param attention_layer:
+            The attention layer to use in the transformer layer.
+        :param feed_forward_layer:
+            The pointwise feed-forward layer to use in the transformer layer.
         :param hidden_dropout:
             Dropout probabilty to apply after hidden layers.
```
[the remark this answers is not in the document]
Comment (danieldk) [reply curated_transformers/layers/transformer.py:69]: This is incorrect, it's applied to output of both hidden layers (attention and ff).

**Pull request merged**